src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/MaxLengthAttribute.cs

```diff
@@ -87,9 +87,7 @@ public override bool IsValid(object value)
                 }
                 else
                 {
-                    // A cast exception previously occurred if a non-{string|array} property was passed
-                    // in so preserve this behavior if the value does not implement ICollection
-                    length = ((Array)value).Length;
+                    throw new InvalidCastException(SR.Format(SR.MaxLengthAttribute_InvalidValueType, value.GetType()));
```

Comment:
What if it is an array? You're removing functionality here which was added to solve a previous issue.

---
In reply to: [121382258](https://github.com/dotnet/corefx/pull/20934#discussion_r121382258) [](ancestors = 121382258)